odoo/models.py

```diff
@@ -2047,58 +2047,122 @@ def _read_group_fill_results(self, domain, groupby, remaining_groupbys,
+        with the domain e.g. ['&', ('date_field', '>=', 'YYYY-08-01'), ('date_field', '<', 'YYYY-11-01')]:
+                                         ___
+                                    ___ |___| ___
+                                    Aug  Sep  Oct
+
+        Minimal filling amount:
+        Using `min_groups`, we can specify that we want at least that amount of contiguous groups.
+        This amount is guaranteed to be provided from `fill_from` if specified, or from the lowest existing
+        group otherwise. This amount is not restricted by `fill_to`. If there is an existing group before
+        `fill_from`, `fill_from` is still used as the starting group for min_groups, because the filling does
+        not apply on that existing group. If neither `fill_from` nor `fill_to` is specified, and there is no
+        existing group, no group will be returned.
+
+        If we set min_groups = 4:
+                                         ___
+                                    ___ |___| ___ ___
+                                    Aug  Sep  Oct Nov
 
         :param list data: the data containing groups
         :param list groupby: name of the first group by
         :param list aggregated_fields: list of aggregated fields in the query
-        :param relativedelta interval: interval between two temporal groups
-                expressed as a relativedelta month by default
+        :param str fill_from: string date(time), inclusive start bound of the fill_temporal range
+        :param str fill_to: string date(time), inclusive end bound of the fill_temporal range
+        :param int min_groups: minimal amount of groups for the fill_temporal range
         :rtype: list
         :return: list
         """
         first_a_gby = annotated_groupbys[0]
-        if not data:
-            return data
         if first_a_gby['type'] not in ('date', 'datetime'):
             return data
         interval = first_a_gby['interval']
+        granularity = first_a_gby['granularity']
+        tz = pytz.timezone(self._context['tz']) if first_a_gby["tz_convert"] else False
         groupby_name = groupby[0]
 
         # existing non null datetimes
-        existing = [d[groupby_name] for d in data if d[groupby_name]]
+        existing = [d[groupby_name] for d in data if d[groupby_name]] or [None]
+        # assumption: existing data is sorted by field 'groupby_name'
+        existing_from, existing_to = existing[0], existing[-1]
+
+        if fill_from:
+            fill_from = date_utils.start_of(odoo.fields.Datetime.to_datetime(fill_from), granularity)
+            if tz:
+                fill_from = tz.localize(fill_from)
+        elif existing_from:
+            fill_from = existing_from
+        if fill_to:
+            fill_to = date_utils.start_of(odoo.fields.Datetime.to_datetime(fill_to), granularity)
+            if tz:
+                fill_to = tz.localize(fill_to)
+        elif existing_to:
+            fill_to = existing_to
+        else:
```

Comment:
I would split fill_from / fill_to first computation from "guess based on missing parameters", aka
```
if fill_from:
    # blah
elif existing_from:
    # blah
if fill_to:
    # blah
elif existing_to:
    # blah

if not fill_to and fill_from:
    fill_to = fill_from
if not fill_from and fill_to:
    fill_from = fill_to
if not fill_from and not fill_to:
    return data
```

That seems clearer to me (as I had to read twice code to understand it ^^)